src/renderers/dom/client/utils/createObjectWithProto.js

```diff
@@ -0,0 +1,33 @@
+/**
```

Comment:
It's not a polyfill, because there is no accurate polyfill for `Object.create`; that's why we're in this predicament. :) The `Object.create` es5-sham (which I'm a contributor on) is over 100 lines.  Setting a proto, on the other hand, is something every JS framework has to do at one point or another, since long before `Object.create` existed.

Can you elaborate on the ES6 class idea?  Do you mean rewriting `SyntheticEvent` to use ES6 classes and real inheritance?  The main problem with that approach is that Babel's own `_inherits` helper uses `Object.create`! Babel helpers in general are not IE8-compatible, so "just use Babel" does not apply directly to the current problem, which is making React work in IE8 without relying on shams in the environment.  (The right solution to using Babel in IE8 without shams, which is what we do in Meteor, is to override the helpers with IE8-compatible versions.)  Also, this would be the first use of the `class` keyword in the entire codebase, which would pull in `_inherits` and other Babel helpers for the first time, and while there is nothing wrong with that, it seems like a big move that I would rather trust to a core contributor, and not necessarily something to block this issue on.

At the end of the day, setting a proto is 3 lines of JS, or one line with `Object.create`, but the latter requires that something called `Object.create` exist in the global environment.  There are logically only a few possible ways to avoid calling global `Object.create` with a small localized code change -- don't use `Object.create` at all; test for it and use it if available; factor out the functionality you need into a utility -- and any of them is acceptable to me.

Since `SyntheticEvent.augmentClass` is already basically a one-off inheritance mechanism, in my opinion it would be reasonable to simply inline the code of `createObjectWithProto` (which either calls `Object.create` or does the equivalent work) into that function until such time as `SyntheticEvent` is moved over to real inheritance.  What do you think?